File: src/trees/traversal/TreeTraversal.java
package src.trees.traversal;

import src.trees.TreeNode;

import java.util.ArrayList;

public interface TreeTraversal {
    ArrayList<Integer> preOrderTraversal(TreeNode root);
    ArrayList<Integer> inOrderTraversal(TreeNode root);
    ArrayList<Integer> postOrderTraversal(TreeNode root);
}

File: src/trees/traversal/TreeTraversal.kt
package src.trees.traversal

import src.trees.TreeNode

interface TreeTraversal {
    fun preOrderTraversal(root: TreeNode?): List<Int>
    fun inOrderTraversal(root: TreeNode?): List<Int>
    fun postOrderTraversal(root: TreeNode?): List<Int>
}
